### PR TITLE
Supporting comments/strings through grammar rewriting

### PR DESCRIPTION
**Pull request by ptersilie**
Replaced the previous "AnySymbol"-approach with a new approach that rewrites the grammars to allow the use of multiline structures such as strings and comments while keeping changes to the parser minimal.

### CODE REVIEW

File: lib/eco/grammar_parser/bootstrap.py [hunk 1]
```diff
@@ -149,21 +190,45 @@ def parse_precedence_symbols(self, symbol):
...
+                    r.add_alternative([])
+                    self.rules[r.symbol] = r
+                # remove whitespace before special rule from parent rule, e.g.
+                # multistring ::= "MLS" WS *match_until "MLS" WS
+                #                       ^ this WS causes shift/reduce conflicts
+                prule = self.rules[Nonterminal(parentrule)]
+                for a in prule.alternatives:
+                    for i in range(len(a)):
+                        sym = a[i]
+                        if sym.name == "WS":
```
Comment (ltratt) [inline lib/eco/grammar_parser/bootstrap.py:213]: Can this not be `if len(a) > i + 1` and remove the `try ... except`?

File: lib/eco/grammar_parser/bootstrap.py [hunk 2]
```diff
@@ -256,14 +319,34 @@ def parse_symbol(self, node):
...
+        s = []
+        for c in symbol.children:
+            if c.symbol.name == ",":
+                continue
+            if c.symbol.name == "WS":
+                continue
+            if c.symbol.name == "f_args":
+                rec_s = self.parse_fargs(symbol.children[0])
+                s.extend(rec_s)
+            if c.lookup == "terminal":
```
Comment (ltratt) [inline lib/eco/grammar_parser/bootstrap.py]: Can the `c.symbol.name` and `c.lookup` clauses both be true?

**Comment by ltratt** [comment]
Only a few minor comments from me...

**Comment by ptersilie** [comment]
@ltratt i think this would be ready to merge

**Comment by ltratt** [comment]
Did you rebase? That makes it really difficult to compare what's changed.

**Comment by ptersilie** [comment]
Yes, I rebased. But you already checked this repo. I just applied your comments and then rebased.

**Comment by ltratt** [comment]
So I have one, potentially minor, comment (see above).

**Comment by ptersilie** [comment]
You are right. Fixed.

**Comment by ltratt** [comment]
OK, so let's squash, then I can merge.

**Comment by ptersilie** [comment]
Squashed.

**Comment by ltratt** [comment]
Thanks!

**Pull request merged**